uta_tools/uta_tools.py

```diff
@@ -42,3 +45,392 @@ def __init__(self, seqrepo_data_path: str = SEQREPO_DATA_PATH,
+            # convert this to the correct accession version
+            chromosome = int(chromosome)
+        except ValueError:
+            # Check if valid accession is given
+            if not await self.uta_db.validate_genomic_ac(chromosome):
+                logger.warning(f"Chromosome is not valid: {chromosome}")
+                return None
+
+        if isinstance(chromosome, str):
+            # Accession given
+            genes_alt_acs = await self.uta_db.chr_to_gene_and_accessions(
+                chromosome, pos, strand=strand,
+                alt_ac=chromosome
+            )
+        else:
+            # Number given
+            genes_alt_acs = await self.uta_db.chr_to_gene_and_accessions(
+                chromosome, pos, strand=strand, alt_ac=None
+            )
+        gene_alt_ac = self._get_gene_and_alt_ac(genes_alt_acs, gene)
+        if not gene_alt_ac:
+            return None
+        gene, alt_ac = gene_alt_ac
+
+        if transcript is None:
+            await self._set_mane_genomic_data(params, gene, alt_ac, pos,
+                                              strand, is_start)
+        else:
+            params["transcript"] = transcript
+            params["gene"] = gene
+            params["pos"] = pos
+            params["chr"] = alt_ac
+            await self._set_genomic_data(params, strand, is_start)
+
+        if residue_mode.lower().strip() == ResidueMode.RESIDUE and is_start:
+            params["pos"] -= 1
+        return TranscriptExonData(**params)
+
+    def _get_gene_and_alt_ac(self,
+                             genes_alt_acs: dict,
+                             gene: Optional[str]) -> Optional[Tuple[str, str]]:
+        """Return gene genomic accession
+
+        :param dict genes_alt_acs: Dictionary containing genes and
+            genomic accessions
+        :return: [Gene, Genomic accession] if both exist
+        """
+        alt_acs = genes_alt_acs["alt_acs"]
+        len_alt_acs = len(alt_acs)
+        if len_alt_acs > 1:
+            logger.warning(f"Found more than one accessions: {alt_acs}")
+            return None
+        elif len_alt_acs == 0:
+            logger.warning("No accessions found")
+            return None
+        alt_ac = next(iter(alt_acs))
+
+        genes = genes_alt_acs["genes"]
+        len_genes = len(genes)
+        input_gene = gene
+        output_gene = None
+        if len_genes == 1:
+            output_gene = next(iter(genes))
+        elif len_genes > 1:
+            logger.warning(f"Found more than one gene: {genes}")
+            return None
+        elif len_genes == 0:
+            logger.warning("No genes found")
+            return None
+
+        if input_gene is not None:
+            input_gene = input_gene.upper()
+            if gene and gene != input_gene:
+                logger.warning(f"Input gene, {input_gene}, does not match "
+                               f"output gene, {output_gene}")
+                return None
```

Comment:
I might be misreading this, but I think the first part of the`if gene` clause is unnecessary because `input_gene` was just `gene` as of line 275, so if `input_gene` isn't None then `gene` can't be None

Also why this mean that input_gene doesn't match output_gene?

Reply:
I messed up the if condition on 277. It should be if `output_gene != input_gene`